### ISSUE TITLE
Running the code

### ISSUE THREAD

**Issue by bluetooth12** [smasherprog/screen_capture_lite#136]
A couple questions:
1. Can your code capture one or all monitors connected to the computer? I am running it now and it shows me a section of my monitor (but not the entire monitor). Is that because of the tiny_jpeg files?
2. Can your code capture the mouse in the frame? Right now the code is generating pictures of mice but not the mouse relative to the rest of the frame on the monitor.

**Comment by bluetooth12** [comment]
3. Why do you have ScreenCapture_C_API.h? Why not write in pure C++? Why do you have a bunch of wrappers?

**Comment by smasherprog** [comment]
The example code captures the while monitor, parts of the monitor. I try to run through a couple of scenarios. 
The base case
https://github.com/smasherprog/screen_capture_lite/blob/47defa25fd20048874f78b230a2bed17777eade2/Example_CPP/Screen_Capture_Example.cpp#L408

Should get you the whole monitor

**Comment by smasherprog** [comment]
Yes thats correct. The mouse image is separate. Otherwise you would need to receive and handle an entire image of everything just because the mouse moved. So your application should combine the mouse on top of the thing its capturing. 

Can you post an example of what you mean when u say, you get a bunch of parts of your monitor?

**Comment by bluetooth12** [comment]
I ran your code and got 95 images. Here are the first 4. The first image is the full monitor. The rest of them are pieces of my monitor.

![0MONITORDIF_](https://user-images.githubusercontent.com/8892431/176724373-37c732bd-8e5f-4ab8-88b8-3af367bf583f.jpg)
![1MONITORDIF_](https://user-images.githubusercontent.com/8892431/176724385-d2212ebc-c2d9-406a-87c5-76d5e059f11a.jpg)
![2MONITORDIF_](https://user-images.githubusercontent.com/8892431/176724389-e29a0a14-cd2f-430a-b060-e508eacd7384.jpg)
![3MONITORDIF_](https://user-images.githubusercontent.com/8892431/176724391-df5466e8-b046-46c8-8453-2a4aa30d30b7.jpg)

**Comment by smasherprog** [comment]
Perfect so thats correct. There are 3 callbacks
  })->onFrameChanged([&](const SL::Screen_Capture::Image& img, const SL::Screen_Capture::Monitor& monitor) {
  
  })->onNewFrame([&](const SL::Screen_Capture::Image& img, const SL::Screen_Capture::Monitor& monitor) {
  
  })->onMouseChanged([&](const SL::Screen_Capture::Image* img, const SL::Screen_Capture::MousePoint &mousepoint) {
  
  })->start_capturing();



ICaptureConfiguration::onNewFrame: This will call back when a new frame is ready on the interval specified in SCL_SetFrameChangeInterval

ICaptureConfiguration::onFrameChanged: This will call back when differences are detected between the last frame and the current one. This is usefull when you want to stream data that you are only sending what has changed, not everything!

ICaptureConfiguration::onMouseChanged: This will call back when the mouse has changed location or the mouse image has changed up to a maximum rate specified in SCL_SetMouseChangeInterval

**Comment by smasherprog** [comment]
The idea with the pieces is that you send only those and you can stitch them together on the other side. If you know that you are going to send everything ALL the time, then just use the OnNewFrame, if you know you are going to be streaming something that doesnt change all too often, then use the OnFrameChanged..

Both can be used at the same time, these are here solely for optimizations

**Comment by smasherprog** [comment]
Sending a 4k res pic on at a decent framerate will generate significant network throughput even with good compression. So, be carefull
